Quick_Deploy/HuggingFaceTransformers/build_server.py

```diff
@@ -0,0 +1,189 @@
+# Copyright 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
```

Comment:
could we have a docker file with build arguments for the tutorial instead of generating one via script?

I can see a separate script for 'generate_mode_repository.py' that would take in list of hf models and build out the model repository

then docker run script to just run.

Guess I'm thinking tutorial wise - may be easier to grok as seperate with docker build and docker run being bash.

Theoretically:

create_repository.py
Dockerfile
build.sh
run.sh

Reply:
We can definitely do it this way. My reasoning to have everything build in a single script was based on "one-click-deployment". I was hoping a single script would abstract users from all the files and details that go into running Triton. If they wanted to look deeper into the deployment mechanisms, the build script wouldn't delete any of the files it created unless there is a re-run.

I can change the script to accept a list of hf models. I'll try to find some more general models since some require special handling and won't be compatible with our "base model" scripts.

Since the docker command will be one-liners, are we okay following the same format as the [vllm](https://github.com/triton-inference-server/tutorials/tree/main/Quick_Deploy/vLLM#step-1-build-a-triton-container-image-with-vllm) tutorial as opposed to creating sh files?

Reply:
Nice work! docker one liners fine and preferred in many scenarios.